### ISSUE TITLE
feature: follow link in split

### ISSUE THREAD

**Issue by mrossinek** [nvim-neorg/neorg#131]
### Discussed in https://github.com/nvim-neorg/neorg/discussions/129

<div type='discussions-op-text'>

<sup>Originally posted by **rjahrj** September 26, 2021</sup>
After finally learning how to follow a mark I am lost on going back? Is there a keybinding for that or a preferred method. and is there a way to possibly follow a mark in a split? </div>

The suggestion was made to support following links in splits (be that horizontal or vertical). I think this can be easily implemented as an optional argument on `core.norg.esupports.goto_link` and we could then provide various keybindings to do this.

**Comment by danymat** [comment]
Okay guys I don't know about the feasibility of this, but I somewhat created this feature for GTD by using buffer variables.

In GTD, I get all the tasks and their corresponding nodes in the AST. That allows me to populate the views with the tasks, and putting the nodes on their respective lines:

```
- Task 1 -- here I set a buffer variable with { line_nr = node1 }
- Task 2 -- here I set a buffer variable with { line_nr = node2 }
```

This is the code responsible for creating, updating, deleting and doing operations on them (go to location, press tab to show more details, ...):

https://github.com/nvim-neorg/neorg/blob/dab516d10a213282c8a186b6f0fba56191fcb7c4/lua/neorg/modules/core/gtd/ui/displayers.lua#L419-L583

**Comment by mrossinek** [comment]
> Okay guys I don't know about the feasibility of this, but I somewhat created this feature for GTD by using buffer variables. This is somewhat OP because we can potentially move to any node in any buffer.
> 
> In GTD, I get all the tasks and their corresponding nodes in the AST. That allows me to populate the views with the tasks, and putting the nodes on their respective lines:

This is interesting. But this seems to be a rather specific usecase for when you're first building out the buffer contents (at which time you can place these buffer variables).
I would really imagine a much simpler approach (like an optional argument to `goto_link`) which specifies that you can open the target in a window split (vertical, horizontal, or an existing window id).

**Comment by mrossinek** [comment]
As I recall this was part of the `new-links` branch and, thus, should now be available. @vhyrro is there an easy way to map a keybind to this functionality? If yes, I guess we can close this issue.

**Comment by vhyrro** [comment]
Yup! This feature is entirely implemented now thanks to all the new goodies from new-links.
You can hit Alt + \<CR\> to open up a link location in a vertical split.

The code that's responsible for this can be found [here](https://github.com/nvim-neorg/neorg/blob/main/lua/neorg/modules/core/keybinds/default_keybinds.lua#L33-L34)!

I'll close this issue now since there's nothing left to do here

**Comment by danymat** [comment]
I don't think

**Comment by danymat** [comment]
I don't think it was the idea we had with @mrossinek, which was to hop in a location that is located in an other split. (Not creating a split when following a location)

**Comment by mrossinek** [comment]
> I don't think it was the idea we had with @mrossinek, which was to hop in a location that is located in an other split. (Not creating a split when following a location)

No @vhyrro is exactly right. I don't quite understand what you mean.

**Comment by vhyrro** [comment]
K, let's do it this way. Dany explained to me what he meant on discord, but I feel like then the title for this issue is a little misleading.
We can close down this issue, since this issue fixes what we meant to fix. We can create a new issue for what @danymat meant later on :)
